src/codegen/formatted_elz/mod.rs

```diff
@@ -1,354 +1,188 @@
-            ExprVariant::ClassConstruction(name, fields_init) => {
-                let mut s = "".to_string();
-                s.push_str(name);
-                s.push_str("{");
-                let fields: Vec<FormatFieldInit> = fields_init
-                    .into_iter()
-                    .map(|(field_name, field_init)| {
-                        FormatFieldInit(field_name.clone(), field_init.clone())
-                    })
-                    .collect();
-                concat_with_separator(&mut s, &fields, ", ", level);
-                s.push_str("}");
-                s
-            }
+fn add_indent(tab: i32) -> String {
+    let mut count = 0i32;
+    let mut s = String::from("");
+    loop {
+        count += 1;
+        if count > tab {
+            break;
         }
+        s.push_str("  ");
     }
+    s
 }
 
-struct FormatFieldInit(String, Expr);
-
-impl FormattedElz for FormatFieldInit {
-    fn formatted_elz(&self, level: usize) -> String {
-        format!("{}: {}", self.0, self.1.formatted_elz(level))
-    }
-}
-
-impl FormattedElz for Argument {
-    fn formatted_elz(&self, level: usize) -> String {
-        let mut s = "".to_string();
-        s.push_str(
-            self.name
-                .as_ref()
-                .map_or_else(|| "".to_string(), |name| format!("{}: ", name))
-                .as_str(),
-        );
-        s.push_str(self.expr.formatted_elz(level).as_str());
-        s
-    }
-}
-
-impl FormattedElz for Operator {
-    fn formatted_elz(&self, _level: usize) -> String {
-        use Operator::*;
-        match self {
-            Plus => format!("+"),
-        }
+fn add_blank(c: &str, pre: &str, next: &str) -> String {
+    let mut s = String::from("");
+    if pre != " " {
+        s.push_str(" ");
     }
-}
-
-impl FormattedElz for ParsedType {
-    fn formatted_elz(&self, level: usize) -> String {
-        use ParsedType::*;
-        match self {
-            TypeName(name) => name.clone(),
-            GenericType {
-                name,
-                type_parameters,
-            } => {
-                let mut s = String::new();
-                s.push_str(name.as_str());
-                s.push_str("[");
-                concat_with_separator(&mut s, type_parameters, ", ", level);
-                s.push_str("]");
-                s
-            }
-        }
+    s.push_str(c);
+    if next != " " {
+        s.push_str(" ");
     }
+    s
 }
 
-// helpers
-fn concat_with_separator<T: FormattedElz>(
-    s: &mut String,
-    vector: &Vec<T>,
-    separator: &'static str,
-    level: usize,
-) {
-    for (index, elem) in vector.iter().enumerate() {
-        let elem_str = elem.formatted_elz(level);
-        s.push_str(elem_str.as_str());
-        if index < vector.len() - 1 {
-            if elem_str != "".to_string() {
-                s.push_str(separator);
-            }
-        }
+fn check_symbol_behind_space(symbol: &str) -> bool {
+    if symbol == ";" {
+        return true;
```

Comment:
In Rust, an exhaust if-else block doesn't need a return keyword.

You can use {true} to get same result.